src/components/input-time-picker/input-time-picker.tsx

```diff
@@ -133,9 +140,10 @@ export class InputTimePicker
    * BCP 47 language tag for desired language and country format
    *
    * @internal
+   * @deprecated set the global `lang` attribute on the element instead.
+   * @mdn [lang](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/lang)
    */
-  @Prop({ attribute: "lang", mutable: true }) locale: string =
-    document.documentElement.lang || navigator.language || "en";
+  @Prop({ mutable: true }) locale: string;
```

Comment:
Removing the `attribute: "lang"` portion here is causing a regression where the input's value isn't relocalizing when the `lang` attribute is being set.  I updated the demo file to use `setAttribute("lang", newLang)` (which we should probably do as part of this PR) and it fails to relocalize the input's value because the `localeWatcher` isn't firing unless you tell Stencil to have it watch the lang attribute.  There might be another way to solve this, but that's just something I noticed from testing this locally.

Reply:
Basically I did this and now clicking the locales on the Input Time Picker's demo page no longer update the visible value in the inputs:

![image](https://user-images.githubusercontent.com/821864/190476920-8cb2fc51-0550-48ac-b565-4406c7bc0c0c.png)

Since `locale` is deprecated, we should definitely update this demo page to use setAttribute, but we'll want to make sure that works with the changes in this PR too.

Reply:
Since our global attributes util only updates a component's `globalAttributes` property, one thought perhaps is maybe we could add a feature to `updateGlobalAttributes` where it can take in an additional prop as an argument for each watched global attr (in `lang`'s case, this would be `locale`) that it can sync the new value in addition to `component.globalAttributes`.  That would negate the need for this additional `attribute: "lang"` argument to the prop decorator and still allow for the `localeWatcher` to fire.

Reply:
Forgot, we're also going to need to fix this in `time-picker` as well, since that component uses this same pattern.

Reply:
Good catch. I think this should be fixed with the work that's going on with the built-in localization (mentioned above).

Reply:
@eriklharper Pushed a fix. Can you take another look?

Reply:
Looks excellent, works with both setAttribute and setting the property directly :)